src/python/pants/engine/internals/native.py

```diff
@@ -188,11 +189,20 @@ def override_thread_logging_destination_to_just_stderr(self):
     def match_path_globs(self, path_globs: PathGlobs, paths: Iterable[str]) -> bool:
         return cast(bool, self.lib.match_path_globs(path_globs, tuple(paths)))
 
-    def nailgun_server_await_bound(self, scheduler, nailgun_server) -> int:
-        return cast(int, self.lib.nailgun_server_await_bound(scheduler, nailgun_server))
+    def nailgun_server_await_bound(self, nailgun_server) -> int:
```

Comment:
Can you put a type anno on `nailgun_server`?

Reply:
I can't... the type is unnameable right now because it's `self.lib.PyNailgunServer` rather than something importable.

I think that we can fix that (I'm not sure that we need to use `importlib` anymore?), and I thought about fixing it here, but it was off topic.